### PR TITLE
Firecert 2186

### PR DESCRIPTION
**Pull request by anjalimukundan**

### CODE REVIEW

File: cypress/support/step_definitions/validations.js [hunk 1]
```diff
@@ -313,7 +313,33 @@ Then(/'(.+)' will (be|stay) in '(.+)' state/, (app, condition, state) => {
         : app;
   const isEventsExpected = condition == CONSTANTS.STAY ? false : true;
   const appObject = UTILS.getEnvVariable(appId);
-  cy.validateLifecycleState(appObject.getAppObjectState().state, appId);
```
Comment (neeradanelxsi) [inline cypress/support/step_definitions/validations.js:316]: can you add comments here

File: cypress/support/step_definitions/validations.js [hunk 2]
```diff
@@ -313,7 +313,33 @@ Then(/'(.+)' will (be|stay) in '(.+)' state/, (app, condition, state) => {
...
+      } else {
+        assert(
+          false,
+          `Expected validationObject to be of "custom" type. Current value : ${Cypress.env(CONSTANTS.COMBINEVALIDATIONOBJECTSJSON)[validationObject].data[0].type}`
+        );
+      }
+    }
+  } else {
+    cy.validateLifecycleState(appObject.getAppObjectState().state, appId);
+  }
```
Comment (neeradanelxsi) [inline cypress/support/step_definitions/validations.js:342]: validateLifecycleHistoryAndEvents should also be under else condition in line 340

File: cypress/support/step_definitions/validations.js [hunk 3]
```diff
@@ -313,7 +313,33 @@ Then(/'(.+)' will (be|stay) in '(.+)' state/, (app, condition, state) => {
...
+      if (
+        Cypress.env(CONSTANTS.COMBINEVALIDATIONOBJECTSJSON)[validationObject].data[0].type ==
+        'custom'
+      ) {
+        const validationObjectData = Cypress.env(CONSTANTS.COMBINEVALIDATIONOBJECTSJSON)[
+          validationObject
+        ].data[0];
+        cy.customValidation(validationObjectData).then((response) => {
+          cy.log('Response from configModule for customValidation ');
+        });
```
Comment (neeradanelxsi) [inline cypress/support/step_definitions/validations.js:332]: is it necessary to have a response for custom validation ?

File: cypress/support/step_definitions/validations.js (hunk 1 above)
Comment (anjalimukundan) [reply cypress/support/step_definitions/validations.js:316]: added comments

File: cypress/support/step_definitions/validations.js (hunk 2 above)
Comment (anjalimukundan) [reply cypress/support/step_definitions/validations.js:342]: moved the code to else condition

File: cypress/support/step_definitions/validations.js (hunk 3 above)
Comment (anjalimukundan) [reply cypress/support/step_definitions/validations.js:332]: Added it for confirming the customValidation response.
Removed the response code.

**Review by jbigel — DISMISSED** [review]

**Review by neeradanelxsi — APPROVED** [review]

**Pull request merged**